src/box/gc.c

```diff
@@ -744,6 +822,411 @@ gc_consumer_update(struct gc_consumer *consumer, const struct vclock *vclock)
 		gc_schedule_cleanup();
 }
 
+void
+gc_consumer_update(struct gc_consumer *consumer, const struct vclock *vclock)
+{
+	if (consumer->is_persistent) {
+		vclock_copy(&consumer->volatile_vclock, vclock);
+		/*
+		 * Do not wake up the background fiber if it's persisting
+		 * consumers right now because it may be waiting for a cbus
+		 * message, which doesn't tolerate spurious wakeups.
+		 */
+		if (gc.persist_scheduled++ == gc.persist_completed)
+			fiber_wakeup(gc.persist_fiber);
```

Comment:
I don't think we need the two counters because the spurious wakeup issue should have been fixed, see #6506. A boolean flag would be enough. It looks like we should drop `cleanup_scheduled`, `cleanup_completed`, as well.

Reply:
Done.